NDTensors/src/lib/SparseArrayDOKs/src/sparsearraydok.jl

```diff
@@ -104,3 +104,28 @@ SparseArrayDOK{T}(a::AbstractArray) where {T} = SparseArrayDOK{T,ndims(a)}(a)
 function SparseArrayDOK{T,N}(a::AbstractArray) where {T,N}
   return SparseArrayInterface.sparse_convert(SparseArrayDOK{T,N}, a)
 end
+
+function Base.resize!(a::SparseArrayDOK{<:Any,N}, new_size::NTuple{N,Integer}) where {N}
+  a.dims = new_size
+  return a
+end
+
+function setindex_maybe_grow!(
+  a::SparseArrayDOK{<:Any,N}, value, i1::Int, I::Int...
```

Comment:
```suggestion
  a::SparseArrayDOK{<:Any,N}, value, index::Vararg{Int,N}
```
to make sure it matches the number of dimensions of the array.